### DISCUSSION TITLE
DeprecationWarning: Changing state of started or joined application is deprecated

### DISCUSSION THREAD

**Discussion by fabianbuettner** [aio-libs/aiohttp#6804] [Q&A] [answered]
Hi,

I am currently in the process of writing a application where I wanted to change the state of a variable which I had put in the app object(app["state"], indicating whether the application is running, stopped, paused, etc.). But each time I modify the state variable I get the DeprectationWarning mentioned in the title.

What is the recommended way of doing that?

best regards,
Fabian

**Comment by Dreamsorcerer** [comment, marked as answer]
Well, it only disallows assigning to top-level app object. So, you could just use a mutable object as a simple solution: `app["state"] = {"current": "starting"}`. You could maybe also create a custom class to more precisely represent the state. I think it's sufficiently common to have some kind of mutable object on the app (e.g. a `ClientSession` for making network requests).

**Reply by fabianbuettner** [reply]
I see, thank you for the answer.
I found someone asking the same question on [StackOverflow](https://stackoverflow.com/a/40617733). Unfortunately, I don't have an account to link to your answer.
